loadtest/control/simplecontroller/actions.go

```diff
@@ -586,6 +593,34 @@ func (c *SimpleController) reload(full bool) control.UserStatus {
 	return c.newInfoStatus("page reloaded")
 }
 
+func (c *SimpleController) viewUser() control.UserStatus {
+	team, err := c.user.Store().RandomTeam()
+	if err != nil {
+		return c.newErrorStatus(err)
+	}
+	channel, err := c.user.Store().RandomChannel(team.Id)
+	if err != nil {
+		return c.newErrorStatus(err)
+	}
+
+	err = c.user.GetChannelMembers(channel.Id, 0, 100)
+	if err != nil {
+		return c.newErrorStatus(err)
+	}
+
+	member, err := c.user.Store().RandomChannelMember(channel.Id)
+	if err != nil {
+		return c.newErrorStatus(err)
+	}
+
+	// GetUsersByIds for that userid
+	_, err = c.user.GetUsersByIds([]string{member.UserId})
+	if err != nil {
+		return c.newErrorStatus(err)
+	}
```

Comment:
Do we need to implement `GetUser` or does the webapp actually call `GetUsersByIds` with one id only?

Reply:
Not sure. My grepping skills are failing me.

```
10:49:41-~/mattermost/mattermost-webapp$git grep "Client4\.getProfilesByIds"
10:49:45-~/mattermost/mattermost-webapp$git grep "Client4\.getUser"
components/suggestion/generic_user_provider.jsx:                    url={Client4.getUsersRoute() + '/' + item.id + '/image?_=' + (item.last_picture_update || 0)}
utils/utils.jsx:    return Client4.getUsersRoute() + '/' + userIdOrObject.id + '/image?_=' + (userIdOrObject.last_picture_update || 0);
utils/utils.jsx:    return Client4.getUsersRoute() + '/' + userId + '/image/default';
```

Reply:
From my search, I think it uses `GetUsersInChannel` and `GetUsersInTeam`. I think `GetUsersInChannel` is the one we are looking for here. But it seems out of scope for this PR ? Perhaps another PR ?

Reply:
> From my search, I think it uses `GetUsersInChannel` and `GetUsersInTeam`. I think `GetUsersInChannel` is the one we are looking for here. But it seems out of scope for this PR ? Perhaps another PR ?

I guess my doubt was about using:

https://godoc.org/github.com/mattermost/mattermost-server/model#Client4.GetUser

as opposed to:

https://github.com/mattermost/mattermost-server/blob/master/model/client4.go#L943

For fetching a single user like we do there.
But it's good for now, just curious.

Reply:
I got that part. My point was that I think the webapp does not fetch a single user at all. It either fetches all users in a channel, or all users in a team. That's what I gleaned.